pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -553,9 +555,8 @@ public void sendAsync(Message<?> message, SendCallback callback) {
                 synchronized (this) {
                     // Update the message metadata before computing the payload chunk size
                     // to avoid a large message cannot be split into chunks.
-                    final long sequenceId = updateMessageMetadataSequenceId(msgMetadata);
-                    String uuid = totalChunks > 1 ? String.format("%s-%d", producerName, sequenceId) : null;
-
+                    final long sequenceId = updateMessageMetadataSequenceId(msgMetadata, chunkId > 0);
+                    String uuid = totalChunks > 1 ? String.format("%s-%s", producerName, chunkUUid) : null;
```

Comment:
- Why do we need to use uuid here? isn't `String.format("%s-%d", producerName, sequenceId)` unique already for each chunk?
- do we need to worry about any uuid collision?

Reply:
Yeah, we need to worry about the uuid collision.
So we should sequence ID to generate uuid.
```
String uuid = totalChunks > 1 ? String.format("%s-%s", producerName,  sequenceId - chunkId) : null;
```